data/npc/lib/npc.lua

```diff
@@ -56,68 +56,44 @@ function doNpcSellItem(cid, itemid, amount, subType, ignoreCap, inBackpacks, bac
 end
 
 local func = function(cid, text, type, e, pcid)
-	if isPlayer(pcid) then
-		doCreatureSay(cid, text, type, false, pcid, getCreaturePosition(cid))
-		e.done = TRUE
+	if Player(pcid):isPlayer() then
+		local creature = Creature(cid)
+		creature:say(text, type, false, pcid, creature:getPosition())
+		e.done = true
 	end
 end
 
 function doCreatureSayWithDelay(cid, text, type, delay, e, pcid)
-	if isPlayer(pcid) then
-		e.done = FALSE
+	if Player(pcid):isPlayer() then
+		e.done = false
 		e.event = addEvent(func, delay < 1 and 1000 or delay, cid, text, type, e, pcid)
 	end
 end
 
-function doPlayerTakeItem(cid, itemid, count)
-	if getPlayerItemCount(cid,itemid) < count then
-		return false
-	end
-
-	while count > 0 do
-		local tempcount = 0
-		if isItemStackable(itemid) then
-			tempcount = math.min (100, count)
-		else
-			tempcount = 1
-		end
-
-		local ret = doPlayerRemoveItem(cid, itemid, tempcount)
-		if ret ~= false then
-			count = count - tempcount
-		else
-			return false
-		end
-	end
-
-	if count ~= 0 then
-		return false
-	end
-	return true
-end
```

Comment:
Do we need that in compat.lua now? Could it be used outside this file?

Reply:
Doesn't function player:removeItem() the same as this?

Reply:
That's exactly my point, if they are the same, are you providing a compatibility alias/shim wrapper for scripts that exist somewhere and may be missing this function?

Reply:
yeah @luanluciano93 add doPlayerTakeItem to compat.lua file returning player:removeItem()